### PR TITLE
Docker: Moved from Alpine Linux to Debian Buster

### PR DESCRIPTION
**Pull request by alt-glitch**
As mentioned in #3712, I've changed the Dockerfile to use python:3.9-buster. Some changes were required since the packages and user/group management are different in Alpine and Debian.

Docker Image was built and tested to be working on Manjaro Linux 5.10.15-1-MANJARO x86_64 20.2.1 Nibia

### CODE REVIEW

File: mitmproxy/tools/web/webaddons.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def open_browser(url: str) -> bool:
     browsers = (
         "windows-default", "macosx",
         "wslview %s",
-        "x-www-browser %s", "gnome-open %s",
+        "x-www-browser %s", "gnome-open %s", "xdg-open",
```
Comment (Kriechi) [inline mitmproxy/tools/web/webaddons.py:49]: Is this related to the Docker change? I don't think this even works when mitmproxy is running inside a container...

File: release/docker/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
-FROM alpine:3.12
+FROM python:3.9-buster
 
-ENV LANG=en_US.UTF-8
+# ENV LANG=en_US.UTF-8
```
Comment (Kriechi) [inline release/docker/Dockerfile:3]: Why did you remove this?

File: release/docker/Dockerfile [hunk 3]
```diff
@@ -9,26 +9,26 @@ COPY $WHEEL_MITMPROXY /home/mitmproxy/
...
         g++ \
-        libffi \
+        # libffi \
         libffi-dev \
-        libstdc++ \
+        # libstdc++\
         openssl \
-        openssl-dev \
+        libssl-dev \
         python3 \
```
Comment (Kriechi) [inline release/docker/Dockerfile:22]: Since the base image is already based on python3, there is no need to (re-)install the package here.

File: release/docker/Dockerfile [hunk 4]
```diff
@@ -9,26 +9,26 @@ COPY $WHEEL_MITMPROXY /home/mitmproxy/
...
-        python3-dev \
+        # python3-dev \
```
Comment (Kriechi) [inline release/docker/Dockerfile:23]: Please remove all these commented-out lines to keep the file tidy.

**Review by Kriechi — CHANGES_REQUESTED** [review]

File: release/docker/Dockerfile (hunk 2 above)
Comment (alt-glitch) [reply release/docker/Dockerfile:3]: Setting this lang ENV was resulting in some error debug messages. On removing this, it seemed to be working. I believe Alpine Linux needed to set the environment and Debian Buster can work fine without it.

File: release/docker/Dockerfile (hunk 4 above)
Comment (alt-glitch) [reply release/docker/Dockerfile:23]: Will do!

File: mitmproxy/tools/web/webaddons.py (hunk 1 above)
Comment (alt-glitch) [reply mitmproxy/tools/web/webaddons.py:49]: This isn't related to the Docker change but was actually part of the previous PR I made (#4435), which I believe was merged, so I don't know why this shows up as a commit since both my fork and the main branch are on par

Comment (alt-glitch) [reply mitmproxy/tools/web/webaddons.py:49]: https://github.com/mitmproxy/mitmproxy/blob/4bd7b6c4eadeaca712f63e0e73f20bcf6aadbffb/mitmproxy/tools/web/webaddons.py#L49
The main branch has this commit too. 
Apologies for any confusion

**Comment by alt-glitch** [comment]
@Kriechi Could you please review my comments? I realise that my previous PR and current one have been bundled up. But I'll be sure to use different branches for different commits/issues now. 
Apart from that I've added the requested changes.

File: release/docker/Dockerfile (hunk 2 above)
Comment (Kriechi) [reply release/docker/Dockerfile:3]: If it is not needed, please remove it.

File: release/docker/docker-entrypoint.sh [hunk 5]
```diff
@@ -2,7 +2,8 @@
 # WARNING: do not change the shebang - the Docker base image might not have what you want!
 
 set -o errexit
-set -o pipefail
+# set -o pipefail
+# Commented out since the option wasn't working on sh in Debian Buster
```
Comment (Kriechi) [inline release/docker/docker-entrypoint.sh:6]: please remove this if its not needed.

File: release/docker/docker-entrypoint.sh [hunk 6]
```diff
@@ -11,7 +12,7 @@ MITMPROXY_PATH="/home/mitmproxy/.mitmproxy"
 if [[ "$1" = "mitmdump" || "$1" = "mitmproxy" || "$1" = "mitmweb" ]]; then
   mkdir -p "$MITMPROXY_PATH"
   chown -R mitmproxy:mitmproxy "$MITMPROXY_PATH"
-  su-exec mitmproxy "$@"
+  su - mitmproxy -c "$@" # su-exec didn't have a package for Debian Buster, hence su
```
Comment (Kriechi) [inline release/docker/docker-entrypoint.sh:15]: please remove the comment - nobody is remembering that a previous version was using su-exec, or for what reason.

File: mitmproxy/tools/web/webaddons.py (hunk 1 above)
Comment (Kriechi) [reply mitmproxy/tools/web/webaddons.py:49]: Please rebase this PR to have a clean commit history and minimize the diff.

**Review by Kriechi — CHANGES_REQUESTED** [review]
